### ISSUE TITLE
⌨ Mobile menu button does not indicate it expands content

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#13142]
## ⌨ Accessibility Issue

The button to toggle the mobile menu does not indicate that it expands content so AT users are unaware and it erroneously announces it as a group of buttons that only contains one item.

### WCAG 2.1 Criteria

 - WCAG 2.1 - 1.3.1 Info and Relationships (Level A)
    - aria-expanded helps establish this relationship by indicating whether the content is currently visible or hidden.

## 🦋 Expected Behaviour

When focus enters a the mobile menu button, it indicates that is is expandable along with the current state (expanded/collapsed).

## 📋 Steps to Reproduce

1. Navigate to `talent.canada.ca`
2. Turn on a screen reader or open the accessibility tree
3. Select the mobile menu button
4. Observe that it is a standalone button and not associated with the mobile menu or announced as a trigger to expand and collapse the menu

## 📸 Screenshot

<img width="298" alt="Image" src="https://github.com/user-attachments/assets/9aa0b2db-d3df-4762-8dde-76f1959d7935" />

## 🙋‍♀️ Proposed Solution

Convert mobile menu into a proper dialog or, add the following:
 - `aria-expanded={isOpen}`
 - `aria-controls="menuID"`

## ✅ Acceptance Criteria

- [ ] Mobile menu trigger properly associated with the menu
- [ ] Mobile menu trigger announced as collapisble and current astate
- [ ] Mobile menu trigger not announced as a group (unless it includes the notification button as well)
